lib/modulesync/git_service.rb

```diff
@@ -0,0 +1,24 @@
+module ModuleSync
+  # Namespace for Git service classes (ie. GitHub, GitLab)
+  module GitService
+    def self.instantiate(type:, options:)
```

Comment:
This is really a factory. Should the class be named `GitServiceFactory` or is that too Java-y?